src/ophys_etl/modules/mesoscope_splitting/__main__.py

```diff
@@ -94,39 +96,56 @@ def write_out_stitched_full_field_image(
 
     output_path: pathlib.Path
         File to be written out
+
+    logger: Optional[logging.Logger]
+        Logger to record traceback if the process fails
+
+    Notes
+    -----
+    If full field image generation fails, the traceback will
+    be written to the logger, but the overall process will
+    not fail so that the ophys session can continue to be
+    processed in LIMS
     """
 
-    full_field_metadata = ScanImageMetadata(
-            path_to_full_field_tiff)
-    full_field_img = stitch_full_field_tiff(
-            path_to_full_field_tiff)
+    try:
+        full_field_metadata = ScanImageMetadata(
+                path_to_full_field_tiff)
+        full_field_img = stitch_full_field_tiff(
+                path_to_full_field_tiff)
 
-    avg_splitter = AvgImageTiffSplitter(
-            path_to_avg_tiff)
+        avg_splitter = AvgImageTiffSplitter(
+                path_to_avg_tiff)
 
-    with_rois = _insert_rois_into_surface_img(
-        full_field_img=full_field_img,
-        full_field_metadata=full_field_metadata,
-        avg_image_splitter=avg_splitter)
+        with_rois = _insert_rois_into_surface_img(
+            full_field_img=full_field_img,
+            full_field_metadata=full_field_metadata,
+            avg_image_splitter=avg_splitter)
 
-    with h5py.File(output_path, "w") as out_file:
+        with h5py.File(output_path, "w") as out_file:
 
-        out_file.create_dataset(
-            "stitched_full_field",
-            data=full_field_img)
+            out_file.create_dataset(
+                "stitched_full_field",
+                data=full_field_img)
 
-        out_file.create_dataset(
-            "stitched_full_field_with_rois",
-            data=with_rois)
+            out_file.create_dataset(
+                "stitched_full_field_with_rois",
+                data=with_rois)
 
-        out_file.create_dataset(
-            "surface_roi_metadata",
-            data=json.dumps(avg_splitter.raw_metadata).encode('utf-8'))
+            out_file.create_dataset(
+                "surface_roi_metadata",
+                data=json.dumps(avg_splitter.raw_metadata).encode('utf-8'))
 
-        ff_metadata = json.dumps(full_field_metadata.raw_metadata)
-        out_file.create_dataset(
-            "full_field_metadata",
-            data=ff_metadata.encode('utf-8'))
+            ff_metadata = json.dumps(full_field_metadata.raw_metadata)
+            out_file.create_dataset(
+                "full_field_metadata",
+                data=ff_metadata.encode('utf-8'))
+    except Exception:
```

Comment:
Can you write this catch in a way that will at least still let us know what the failure was? Right now I think we would only get the message below while the actual error will be suppressed. Would seem to make debugging a failure difficult.

Reply:
Thanks for clarifying that the `traceback.format_exc()` in the message does what I ask.